### ISSUE TITLE
Comm/Subphase data in stats files will have object temporary IDs

### ISSUE THREAD

**Issue by PhilMiller** [DARMA-tasking/vt#1128]
**Describe the bug**

`NodeStats::startIterCleanup` rewrites its stats storage phase by phase to use permanent IDs instead of temporary IDs so that they're consistent in the dumped stats file regardless of where objects have migrated. This conversion is missing for subphase and communication data, and so those entries in stats files will exist in terms of temporary IDs.

**To Reproduce**
1. Example/test/snippet of code that fails
2. Compiler, platform, libraries
3. Run command: number of processors, threading options, etc.

**Expected behavior**

**Screenshots**

**Platform (please complete the following information):**
 - Supercomputer or desktop [e.g., Trinity, Sierra)
 - Compiler environment [e.g., clang-3.9, gcc8]
 - Libraries [e.g., mpich-2, openmpi)
 - Commit ID or release of VT [e.g., vt-1.0.0]

**Additional context**

**Comment by nlslatt** [comment]
@pbmille If I'm reading the code correctly, it looks like the `CommKeyType` knows both the temporary and permanent IDs of the objects involved and will write to the stats files using permanent IDs (`comm.toObj()` and `comm.fromObj()`).  Do you agree with that?

**Comment by nlslatt** [comment]
@PhilMiller Do you agree that it looks like the subphase comm data is still being allowed to grow without bound in `startIterCleanup`?

**Comment by PhilMiller** [comment]
> @PhilMiller If I'm reading the code correctly, it looks like the `CommKeyType` knows both the temporary and permanent IDs of the objects involved and will write to the stats files using permanent IDs (`comm.toObj()` and `comm.fromObj()`). Do you agree with that?

Yes, your analysis is right

**Comment by PhilMiller** [comment]
> @PhilMiller Do you agree that it looks like the subphase comm data is still being allowed to grow without bound in `startIterCleanup`?

Yes, that should be fixed. The two commits you pushed look good.

**Comment by nlslatt** [comment]
@PhilMiller I didn't write a test for this.  Given that you're planning to refactor to remove temp IDs, do you think this PR is okay to go through without a test attached to it?

**Comment by PhilMiller** [comment]
Yeah, I think it's fine, given the uniformity of that code, the scarce use of the comm data, and the absent use of the subphase comm data as yet.

**Comment by nlslatt** [comment]
> > @PhilMiller Do you agree that it looks like the subphase comm data is still being allowed to grow without bound in `startIterCleanup`?
> 
> Yes, that should be fixed. The two commits you pushed look good.

I'm opening a new issue for the memory growth, #1154, and will handle it in a separate PR.

**Comment by PhilMiller** [comment]
Irrelevant on develop since the elimination of stand-alone temporary object IDs. May still impact stable branch.

**Comment by nlslatt** [comment]
#1418 replaces #1141.

**Comment by cz4rs** [comment]
Closing after #1418 was merged on release-candidate branch. Does not affect `develop`.
